File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int l=0;
        int r=height.length-1;
        int maxwater=0;
        while(l<r){
            int waterlevel=Math.min(height[l],height[r]);
            int width=r-l;
            maxwater=Math.max(maxwater,(waterlevel*width));
            if(height[l]<height[r]){
                l++;
            }
            else{
                r--;
            }
        }
        return maxwater;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l=0
        r=len(height)-1
        maxwater=0
        while(l<r):
            waterlevel=min(height[l],height[r])
            width=r-l
            maxwater=max(maxwater,(waterlevel*width))
            if(height[l]<height[r]):
                l+=1
            else:
                r-=1
        return maxwater;